src/course-user/course-user.service.ts

```diff
@@ -1,4 +1,33 @@
 import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { CourseUserEntity } from '../entities/course-user.entity';
+import { CourseUserRepository } from './course-user.repository';
+import { CourseUserDto } from './dto/course-user.dto';
 
 @Injectable()
-export class CourseUserService {}
+export class CourseUserService {
+  constructor(
+    @InjectRepository(CourseUserRepository) private courseUserRepository: CourseUserRepository,
+  ) {}
+
+  async courseUserExists({ userId, courseId }: CourseUserDto): Promise<CourseUserEntity> {
+    return await this.courseUserRepository.findOne({ userId, courseId });
+  }
+
+  async createCourseUser({ userId, courseId }: CourseUserDto): Promise<CourseUserEntity> {
+    const createCourseUserObject = this.courseUserRepository.create({
+      courseId,
+      userId,
+      completed: false,
+    });
+
+    return await this.courseUserRepository.save(createCourseUserObject);
+  }
+
+  async updateCourseUser({ userId, courseId }: CourseUserDto): Promise<CourseUserEntity> {
```

Comment:
as this is specifically to update the CourseUser (rather than a flexible/generic patch method) lets name this function `completeCourse` or something similar

Reply:
i suggested `getCourseUser` for the first function, currently named `completeCourse`, but changing `updateCourseUser` to `completeCourse`. think the recent name change was an accident!

Reply:
Thanks - The correct refactor has been done now